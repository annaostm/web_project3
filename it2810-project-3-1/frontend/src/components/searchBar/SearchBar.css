.content {
    margin-inline: 20px;
}

.title {
    color: white;
    text-align: center;
    font-size: 32px;
    font-weight: 700;
    margin-top: 50px;
    margin-bottom: 20px;
}

.search-bar {
    border-width: 3px;
    border-radius: 5px;
    height: 45px;
    width: 400px;
    font-size: large;
    border-color: #db0000;
    margin-block: 30px;
    padding-inline: 50px;
}

.filter_buttons {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    
}

.search-bar-field{
    display: grid;
    grid-template-columns: minmax(0, -1fr)Â´;
    grid-gap: 20px;
    margin: 0 auto;
    margin-top: 0px;
    margin-bottom: 30px;
    place-items: center;
    margin-block: 30px;
}

.filter{
    max-width: 1000px;
}

@media only screen and (max-width: 900px) {
    .title {
        font-size: 22px;

    }
    .search-bar {
        height: 45px;
        width: 250px;
    }

    .search-bar-field{
        grid-template-rows: minmax(1, 1fr);
    }

    .filter_buttons {
        flex-direction: column;
    }
}

@media only screen and (max-width: 500px) {
    .title {
        font-size: 18px;
    }
    .search-bar {
        height: 40px;
        width: 200px;
    }
    
}