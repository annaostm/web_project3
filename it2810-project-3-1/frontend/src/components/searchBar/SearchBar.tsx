import './SearchBar.css';
import { useState } from "react";
import DisplayData from "../displayData/DisplayData";
import SelectLabels from "../filterButton/FilterButton";
import "./SearchBar.css";

// Interface for the search bar
interface SearchProps {
  type: string;
}

// SearchBar component that renders the search bar and the filter buttons
const SearchBar = (props: SearchProps) => {
  // State variables for the search bar
  const [input, setInput] = useState<string>("");

  //Options for sorting and filtering buttons
  const sort_by: string[] = ["Title","Oldest", "Newest"];
  const listed_in_movies: string[] = [
    "Action & Adventure",
    "Horror Movies",
    "Sci-Fi & Fantasy",
    "Children & Family Movies",
    "Classic Movies",
    "Comedies",
    "Music & Musicals",
    "Documentaries",
    "Dramas",
    "Romantic Movies",
    "Thrillers",
  ];
  const listed_in_series: string[] = [
    "Anime Series",
    "British TV Shows",
    "Crime TV Shows",
    "Docuseries",
    "Kids' TV",
    "Korean TV Shows",
    "Reality TV",
    "TV Action & Adventure",
    "TV Comedies",
    "TV Dramas",
    "TV Horror",
    "TV Mysteries",
    "International TV Shows",
  ];

  //Sorts the array of genres alphabetically
  listed_in_series.sort();
  listed_in_movies.sort();

  //Renders the search bar and the filter buttons
  return (
    <div className="search-filter-grid">
      <div className="content">
        <div className="title">
          <p>Search for {props.type === "Movie" ? "movies" : "TV shows"}</p>
        </div>
        <div className="search-bar-field">
          <input
            className="search-bar"
            aria-label="Search bar"
            type="text"
            data-cy="search"
            placeholder="Search for entertainment..."
            value={input}
            onChange={(e) => {
              setInput(e.target.value);
            }}
          />
          <div className="filter_buttons">
            <SelectLabels
              options={
                props.type === "TV Show" ? listed_in_series : listed_in_movies
              }
              name={"Filter"}
            />
            <SelectLabels options={sort_by} name={"Sort by"} />
          </div>
        </div>
        {input.length > 0 ? (
          <DisplayData type={props.type} input={input} />
        ) : (
          <DisplayData type={props.type} />
        )}
      </div>
    </div>
  );
};
export default SearchBar;
