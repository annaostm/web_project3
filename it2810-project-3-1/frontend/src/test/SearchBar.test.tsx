import renderer from "react-test-renderer";
import SearchBar from "../components/searchBar/SearchBar";
import "@testing-library/jest-dom/extend-expect";
import { gql } from "@apollo/client";
import { unmountComponentAtNode } from "react-dom";
import { MockedProvider } from "@apollo/client/testing";

// setup a DOM element as a render target

let container: Element | DocumentFragment | null = null;

// setting up mocks for the apollo client

const mocks = [
  {
    request: {
      query: gql`
        query Query {
          movies {
            show_id
            title
            type
            director
          }
        }
      `,
    },
    result: {
      data: {
        movies: [
          {
            show_id: 1,
            title: "My Little Pony: A New Generation",
            type: "TV Show",
            director: "director",
          },
        ],
      },
    },
  },
];

// setting up the mock for the apollo client
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

// cleanup on exiting
afterEach(() => {
  // cleanup on exiting
  if (container == null) {
    throw "container is null";
  }
  unmountComponentAtNode(container);
  container = null;
});

const b = renderer.create(
  <MockedProvider mocks={mocks}>
    <SearchBar type="Movie" />
  </MockedProvider>
);

//Test for SearchBar

it("Test for SearchBar", () => {
  const tree = b.toJSON();
  expect(tree).toMatchSnapshot();
});
