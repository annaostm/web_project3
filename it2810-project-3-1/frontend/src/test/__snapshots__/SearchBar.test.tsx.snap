// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for SearchBar 1`] = `
<div
  className="search-filter-grid"
>
  <div
    className="content"
  >
    <div
      className="title"
    >
      <p>
        Search for 
        movies
      </p>
    </div>
    <div
      className="search-bar-field"
    >
      <input
        aria-label="Search bar"
        className="search-bar"
        data-cy="search"
        onChange={[Function]}
        placeholder="Search for entertainment..."
        type="text"
        value=""
      />
      <div
        className="filter_buttons"
      >
        <div
          className="filter"
        >
          <div
            className="MuiFormControl-root css-1onkz90-MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-9f75lj-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              id="demo-simple-select-helper-label"
            >
              Filter
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1vt7zlm-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-cy="filter-menu"
              onClick={[Function]}
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-helper-label demo-simple-select-helper"
                className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select-helper"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled={false}
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    filter
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="filter"
        >
          <div
            className="MuiFormControl-root css-1onkz90-MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-9f75lj-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              id="demo-simple-select-helper-label"
            >
              Sort by
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1vt7zlm-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-cy="filter-menu"
              onClick={[Function]}
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-helper-label demo-simple-select-helper"
                className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select-helper"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled={false}
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    filter
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <p
      className="wait-text"
    >
      Loading...
    </p>
  </div>
</div>
`;
